test.html

```diff
@@ -54,23 +77,29 @@
 	num_codes = quirc.count(qr);
 	log('num_codes', num_codes);
 	for (i = 0; i < num_codes; i++) {
-	    // struct quirc_code code;
-	    // struct quirc_data data;
-	    // quirc_decode_error_t err;
 
-	    quirc.extract(qr, i, 0); // code
+		quirc.extract(qr, i, image); // code
 
-	    /* Decoding stage */
-	    // err = quirc_decode(&code, &data);
-	    if (err)
-		    console.error("DECODE FAILED: %s\n"); // quirc_strerror(err)
-	    else
-		    console.log("Data: %s\n", data.payload);
-    }
+		/* Decoding stage */
+		err = quirc.decode(image, image + 4096);
+		if (err) {
+			console.error("DECODE FAILED: %d\n", err); // quirc_strerror(err)
+		} else {
+			var version = Module.HEAP32[ (image + 4096) / 4 ];
```

Comment:
The bad way: read struct definition in quirc.h and figure that ints are probably 4 bytes and the struct is probably laid out in mem without gaps. I'm overwriting part of the image to store the extract and decode results.